lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2186,6 +2186,18 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
     return mapValue(BV, transFunction(F));
   }
 
+  case OpExpectINTEL: {
+    IRBuilder<> Builder(BB);
+    SPIRVExpectINTELInstBase *BC = static_cast<SPIRVExpectINTELInstBase *>(BV);
+    Type *RetTy = transType(BC->getType());
+    Value *Val = transValue(BC->getOperand(0), F, BB);
+    Value *ExpVal = transValue(BC->getOperand(1), F, BB);
+    auto IID = Intrinsic::expect;
+    Value *Args[] = {Val, ExpVal};
+    auto *IntrinsicCall = Builder.CreateIntrinsic(IID, RetTy, Args);
+    return mapValue(BV, IntrinsicCall);
```

Comment:
Will it work this way?
```suggestion
    return mapValue(BV, Builder.CreateIntrinsic(Intrinsic::expect, RetTy, {Val, ExpVal}));
```

Reply:
It shall. I just had a desire to have some granularity here (otherwise I would inline Val and ExpVal processing here as well). But yeah, your suggestion looks nicer.